src/application/pokegear/configure/pokegear_configure.c

```diff
@@ -0,0 +1,259 @@
+static int PokegearConfigure_MainTask_FadeOutApp(PokegearConfigureAppData *configureApp);
+
+BOOL PokegearConfigure_Init(OverlayManager *man, int *state) {
+    PokegearAppData *pokegearApp;
+    PokegearConfigureAppData *configureApp;
+
+    pokegearApp = OverlayManager_GetArgs(man);
+    Heap_Create(HEAP_ID_3, HEAP_ID_POKEGEAR_APP, 0x20000);
+    configureApp = OverlayManager_CreateAndGetData(man, sizeof(PokegearConfigureAppData), HEAP_ID_POKEGEAR_APP);
+    memset(configureApp, 0, sizeof(PokegearConfigureAppData));
+    configureApp->pokegear = pokegearApp;
+    configureApp->heapId = HEAP_ID_POKEGEAR_APP;
+    PokegearConfigure_InitInternal(configureApp);
+    return TRUE;
+}
+
+BOOL PokegearConfigure_Main(OverlayManager *man, int *state) {
+    PokegearConfigureAppData *configureApp;
+
+    configureApp = OverlayManager_GetData(man);
+    switch (*state) {
+    case PGCONF_MAIN_STATE_LOAD:
+        *state = PokegearConfigure_MainTask_LoadGFX(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_HANDLE_INPUT:
+        *state = PokegearConfigure_MainTask_HandleInput(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_UNLOAD:
+        *state = PokegearConfigure_MainTask_UnloadGFX(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_CONTEXT_MENU:
+        *state = PokegearConfigure_MainTask_ContextMenu(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_SWAP_SKINS:
+        *state = PokegearConfigure_MainTask_SwapSkins(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_FADE_IN:
+        *state = PokegearConfigure_MainTask_FadeIn(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_FADE_OUT:
+        *state = PokegearConfigure_MainTask_FadeOut(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_FADE_IN_APP:
+        *state = PokegearConfigure_MainTask_FadeInApp(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_FADE_OUT_APP:
+        *state = PokegearConfigure_MainTask_FadeOutApp(configureApp);
+        break;
+    case PGCONF_MAIN_STATE_QUIT:
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+BOOL PokegearConfigure_Exit(OverlayManager *man, int *state) {
+    PokegearConfigureAppData *configureApp;
+    enum HeapID heapID;
+
+    configureApp = OverlayManager_GetData(man);
+    PokegearConfigure_ExitInternal(configureApp);
+    configureApp->pokegear->isSwitchApp = TRUE;
+    heapID = configureApp->heapId;
+    OverlayManager_FreeData(man);
+    Heap_Destroy(heapID);
+    return TRUE;
+}
+
+static void PokegearConfigure_InitInternal(PokegearConfigureAppData *configureApp) {
+    configureApp->pokegear->childAppdata = configureApp;
+    configureApp->pokegear->reselectAppCB = PokegearConfigure_OnReselectApp;
+    configureApp->skin = Pokegear_GetBackgroundStyle(configureApp->pokegear->savePokegear);
+    configureApp->unlockedSkins = Pokegear_GetUnlockedSkins(configureApp->pokegear->savePokegear);
+    configureApp->unlockedSkins = 0xFF; // nani the fuck?
```

Comment:
yeah this is a bug lmao

Reply:
Remove comment as is or change it to reflect the view that it was a removed feature.